code_search/code-search.ipynb

```diff
@@ -0,0 +1,451 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "-"
+    }
+   },
+   "source": [
+    "# Code Search on Kubeflow\n",
+    "\n",
+    "This notebook implements an end-to-end Semantic Code Search on top of [Kubeflow](https://www.kubeflow.org/) - given an input query string, get a list of code snippets semantically similar to the query string. \n",
+    "\n",
+    "\n",
+    "# TODO\n",
+    "\n",
+    "- Appropriate Working Directories for each of the bash commands\n",
+    "- `ks param set` requires a small refactor of the Ksonnet Application.\n",
+    "- Verify everything runs."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Prerequisites\n",
+    "\n",
+    "**NOTE**: If you are using this notebook on a Kubeflow cluster, all the pre-prequisites have already been set up.\n",
+    "\n",
+    "* `Kubeflow v0.2.2`\n",
+    "  This notebook assumes a Kubeflow cluster is already deployed. See [Getting Started with Kubeflow](https://www.kubeflow.org/docs/started/getting-started/).\n",
+    "\n",
+    "* `Python 2.7` (bundled with `pip`) \n",
+    "  For this demo, we will use Python 2.7. This restriction is due to [Apache Beam](https://beam.apache.org/), which\n",
+    "  does not support Python 3 yet (See [BEAM-1251](https://issues.apache.org/jira/browse/BEAM-1251)).\n",
+    "\n",
+    "* `Google Cloud SDK`\n",
+    "  This example will use tools from the [Google Cloud SDK](https://cloud.google.com/sdk/). The SDK must be\n",
+    "  authenticated and authorized. See [Authentication Overview](https://cloud.google.com/docs/authentication/).\n",
+    "  \n",
+    "* `Ksonnet 0.12`\n",
+    "  We use [Ksonnet](https://ksonnet.io/) to write Kubernetes jobs in a declarative manner to be run on top of Kubeflow."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "\n",
+    "echo \"Python Version Info: \" && python2 --version && echo\n",
+    "echo \"Google Cloud SDK Info: \" && gcloud --version && echo\n",
+    "echo \"Kubectl Version Info: \" && kubectl version && echo\n",
+    "echo \"Ksonnet Version Info: \" && ks version"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Get the Source Code\n",
+    "\n",
+    "Let us clone the source code for Code Search from [kubeflow/examples](https://github.com/kubeflow/examples)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "\n",
+    "# NOTE: This cell must only be run once or the clone will complain.\n",
+    "git clone --depth=1 https://github.com/kubeflow/examples"
```

Comment:
Users need to run this step to get the notebook; so they need to run it before they execute the notebook. So this should probably be moved into README.md.

Reply:
I believe it should be in both places since we are not baking the source code in the codelab images.